HLTrigger/Configuration/python/customizeHLTforCMSSW.py

```diff
@@ -39,6 +39,119 @@ def customiseForOffline(process):
 
     return process
 
+def customizeHLTfor47611(process):
+    """ Add CAGeometry ESProducer"""
+    
+    if not hasattr(process, 'HLTRecoPixelTracksSequence'):
+        return process
+        
+    ca_producers = ['CAHitNtupletAlpakaPhase1@alpaka','alpaka_serial_sync::CAHitNtupletAlpakaPhase1']
+
+    ca_parameters = [ 'CAThetaCutBarrel', 'CAThetaCutForward', 
+        'dcaCutInnerTriplet', 'dcaCutOuterTriplet', 
+        'doPtCut', 'doZ0Cut', 'idealConditions', 
+        'includeJumpingForwardDoublets', 'phiCuts','doClusterCut','CPE'] 
+
+    process.hltCAGeometry = cms.ESProducer('CAGeometryESProducer@alpaka',
+        caDCACuts = cms.vdouble(
+            0.15, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25, 0.25
+        ),
+        caThetaCuts = cms.vdouble(
+            0.002,
+            0.002,
+            0.002,
+            0.002,
+            0.003,
+            0.003,
+            0.003,
+            0.003,
+            0.003,
+            0.003
+        ),
+        startingPairs = cms.vint32( [i for i in range(8)] + [13, 14, 15, 16, 17, 18, 19]),
+        pairGraph = cms.vint32( 0, 1, 0, 4, 0,
+            7, 1, 2, 1, 4,
+            1, 7, 4, 5, 7,
+            8, 2, 3, 2, 4,
+            2, 7, 5, 6, 8,
+            9, 0, 2, 1, 3,
+            0, 5, 0, 8, 
+            4, 6, 7, 9 
+        ),
+        phiCuts = cms.vint32( 522, 730, 730, 522, 626,
+            626, 522, 522, 626, 626,
+            626, 522, 522, 522, 522,
+            522, 522, 522, 522
+        ),
+        minZ = cms.vdouble(
+                -20., 0., -30., -22., 10., 
+                -30., -70., -70., -22., 15., 
+                -30, -70., -70., -20., -22., 
+                0, -30., -70., -70.
+        ),
+        maxZ = cms.vdouble( 20., 30., 0., 22., 30., 
+            -10., 70., 70., 22., 30., 
+            -15., 70., 70., 20., 22., 
+            30., 0., 70., 70.),
+        maxR = cms.vdouble(20., 9., 9., 20., 7., 
+            7., 5., 5., 20., 6., 
+            6., 5., 5., 20., 20., 
+            9., 9., 9., 9.),
+        appendToDataLabel = cms.string('hltCAGeometry'),
+        alpaka = cms.untracked.PSet(
+        backend = cms.untracked.string(''),
+        synchronize = cms.optional.untracked.bool
+        )
+    )
+
+    for ca_producer in ca_producers:
+        for prod in producers_by_type(process, ca_producer):
+
+            if not hasattr(prod, 'caGeometry'):
+                setattr(prod, 'caGeometry', cms.string('hltCAGeometry'))
+            
```

Comment:
Why are these (collections of) parameters explicitly listed here in the HLT customisation function, instead of using fillDescriptions for this?

Reply:
These plugins are already existing, so simply delete here in the customisation all old/removed parameters as well as all parameters with changed values, with fillDescriptions then taking care of new values and new parameters.

Reply:
True, let me take care of these (and those above).